src/libraries/icubmod/bcbBattery/bcbBattery.cpp

```diff
@@ -285,50 +197,74 @@ void BcbBattery::run()
         sprintf(buff, "battery status: %+6.1fA   % 6.1fV   charge:% 6.1f%%    time: %s", battery_current, battery_voltage, battery_charge, battery_timestamp);
         yDebug("BcbBattery::run() log_buffer is: %s", buff);
     }
+
+    //flush the buffer
+    pSerial->flush();
 }
 
 bool BcbBattery::getBatteryVoltage(double &voltage)
 {
-    lock_guard<mutex> lck(mtx);
-    voltage = battery_voltage;
+    voltage = batteryReader->battery_voltage;
```

Comment:
As in a previous comment, there is a potential race condition here and in the other ``get*`` below.